crates/rust-analyzer/src/handlers/notification.rs

```diff
@@ -64,7 +66,16 @@ pub(crate) fn handle_did_open_text_document(
         if already_exists {
             tracing::error!("duplicate DidOpenTextDocument: {}", path);
         }
-        state.vfs.write().0.set_file_contents(path, Some(params.text_document.text.into_bytes()));
+
+        let vfs = &mut state.vfs.write().0;
+        vfs.set_file_contents(path, Some(params.text_document.text.into_bytes()));
```

Comment:
Also undo this please, because this will actually widen the temporary scope of the guard compared to the one liner

Reply:
can do.

Reply:
undid this change.